share/ansible_navigator/utils/catalog_collections.py

```diff
@@ -116,62 +115,57 @@ def _generate_checksum(file_path: str, relative_path: str) -> Dict:
     def _process_plugin_dir(
         self,
         plugin_type: str,
-        filenames: List,
-        file_checksums: Dict,
-        dirpath: str,
+        filenames: Generator[Path, None, None],
```

Comment:
Is it wise to only accept generators and not generic iterables here? The code below only relies on the iteration protocol so why limit this to a very narrow iterator subtype?

Reply:
Dunno. I think the rule of thumb is to have a specific return type but broad argument types. But I won't insist. Resolving.